backend/src/hooks/check-has-permission/check-has-permission.hook.ts

```diff
@@ -0,0 +1,34 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+
+import { HttpCode, PermissionKey } from '~/common/enums/enums';
+import { UsersByIdResponseDto } from '~/common/types/types';
+import { PermissionsError } from '~/exceptions/exceptions';
+
+const checkHasPermissions =
+  (...permissions: PermissionKey[]) =>
+  async (req: FastifyRequest, rep: FastifyReply): Promise<void> => {
+    try {
+      const userPermissions =
+        (
+          req.user as UsersByIdResponseDto & {
+            permissions: [];
+          }
+        ).permissions ?? [];
+
+      if (!checkPermissionKeys(permissions, userPermissions)) {
+        throw new PermissionsError();
+      }
+    } catch (err) {
+      rep.code(HttpCode.FORBIDDEN).send(err);
+    }
+  };
+
+const checkPermissionKeys = (
+  requiredPermissions: PermissionKey[],
+  userPermissions: PermissionKey[],
+): boolean =>
+  requiredPermissions.every((permission) => {
+    return userPermissions.includes(permission);
+  });
```

Comment:
```suggestion

const checkHasPermission = (
  permissions: PermissionKey[],
  userPermissions: PermissionKey[],
): boolean => {
  return pagePermissions.every((pagePermission) => {
    return userPermissions.some(
      (userPermission) => userPermission === pagePermission,
    );
  });
};
```
move it to the shared folder. you will need it on the both sides